src/openapi/streaming/subscription.js

```diff
@@ -295,6 +310,11 @@ function Subscription(streamingContextId, transport, serviceGroup, url, subscrip
     this.referenceId = null;
 
     /**
+     * The number of retries this subscription has made
+     */
+    this.retryCounter = 0;

```

Comment:
You should also reset it in subscription.reset method, and clear ongoing retry timeout if present.

I think we should also dispose of retries whenever new subscription request comes in. Especially when user requests unsubscribe.